R/tm_g_ae_sub.R

```diff
@@ -289,40 +313,16 @@ srv_g_ae_sub <- function(id,
     output_q <- reactive({
       ANL <- data[[dataname]]() # nolint
       ADSL <- data[["ADSL"]]() # nolint
-      validate_has_data(ANL, min_nrow = 10)
-      iv_comp <- shinyvalidate::InputValidator$new()
-      iv_comp$add_rule("arm_trt", shinyvalidate::sv_not_equal(
-        input$arm_ref,
-        message_fmt = "Must not be equal to Control"
-      ))
-      iv_comp$add_rule("arm_ref", shinyvalidate::sv_not_equal(
-        input$arm_trt,
-        message_fmt = "Must not be equal to Treatment"
-      ))
-      iv_comp$enable()
-      validate(need(iv_comp$is_valid(), "Misspecification error: please observe red flags in the encodings."))
 
-      validate(need(iv$is_valid(), "Misspecification error: please observe red flags in the encodings."))
+      teal::validate_has_data(ANL, min_nrow = 10, msg = sprintf("%s has not enough data", dataname))
+
+      teal::validate_inputs(iv())
+
       validate(need(
-        is.factor(ANL[[input$arm_var]]),
-        "Selected arm variable needs to be a factor. Contact the app developer."
+        input$arm_trt %in% unique(ANL[[input$arm_var]]) ||
+          input$arm_ref %in% unique(ANL[[input$arm_var]]),
```

Comment:
we `need` both to be present in a data
```suggestion
        input$arm_trt %in% ANL[[input$arm_var]] &&
          input$arm_ref %in% ANL[[input$arm_var]],
```